src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -7199,4 +7199,211 @@ static void Main()
     }
+
+    [Fact]
+    public void ConditionalAccess_01()
+    {
+        var source = CSharpTestSource.Parse("""
+            class C
+            {
+                void M() => throw null!;
+
+                static void Main()
+                {
+                    var c = new C();
+                    c?.M();
+                }
+            }
+            """, "Program.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation(source);
+        var model = comp.GetSemanticModel(source);
+        var node = source.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+        var locationSpecifier = model.GetInterceptableLocation(node)!;
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            #nullable enable
+            using System;
+
+            static class Interceptors
+            {
+                {{locationSpecifier.GetInterceptsLocationAttributeSyntax()}}
+                public static void M1(this C? c) => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var verifier = CompileAndVerify([source, interceptors, s_attributesTree], expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void ConditionalAccess_02()
+    {
+        // use a location specifier which refers to a conditional access that is not being invoked.
+        var source = CSharpTestSource.Parse("""
+            class C
+            {
+                int P => throw null!;
+
+                static void Main()
+                {
+                    var c = new C();
+                    _ = c?.P;
+                }
+            }
+            """, "Program.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation(source);
+        var model = (CSharpSemanticModel)comp.GetSemanticModel(source);
+        var node = source.GetRoot().DescendantNodes().OfType<MemberBindingExpressionSyntax>().Single();
+        var locationSpecifier = model.GetInterceptableLocationInternal(node.Name, cancellationToken: default)!;
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            #nullable enable
+            using System;
+
+            static class Interceptors
+            {
+                {{locationSpecifier.GetInterceptsLocationAttributeSyntax()}}
+                public static void M1(this C? c) => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        comp = CreateCompilation([source, interceptors, s_attributesTree]);
+        comp.VerifyEmitDiagnostics(
+            // Interceptors.cs(6,6): error CS9151: Possible method name 'P' cannot be intercepted because it is not being invoked.
+            //     [global::System.Runtime.CompilerServices.InterceptsLocationAttribute(1, "q2jDXUSFcU71GJHh7313cHEAAABQcm9ncmFtLmNz")]
```

Comment:
>[global::System.Runtime.CompilerServices.InterceptsLocationAttribute(1, "q2jDXUSFcU71GJHh7313cHEAAABQcm9ncmFtLmNz")] [](http://example.com/codeflow?start=19&length=116)

Not related to this PR: Have we considered making `GetInterceptsLocationAttributeSyntax` output a comment with a human-readable location as you suggested source generators should do when writing an `[InterceptsLocation(...)]` attribute?

Reply:
It's possible, but for output which is most appropriate over multiple lines, it seems better to avoid having to handle indentation and so on. (even if generator author doesn't care about unnecessary name qualifiers, they probably care about reasonable indentation and whitespace.)